tests/e2e/vsan_stretched_cluster_utils.go

```diff
@@ -514,3 +519,288 @@ func invokeDockerPauseNKillOnContainerInParallel(sshClientConfig *ssh.ClientConf
+					volumespec.PersistentVolumeClaim.ClaimName, pv.ObjectMeta.Name, sspod.Name)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}
+	}
+	if isDeploymentRequired {
+		framework.Logf("Creating PVC")
+		pvclaim, err := createPVC(client, namespace, nil, diskSize, sc, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		pvclaims = append(pvclaims, pvclaim)
+		persistentvolumes, err := fpv.WaitForPVClaimBoundPhase(client, pvclaims, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		volHandle := persistentvolumes[0].Spec.CSI.VolumeHandle
+		gomega.Expect(volHandle).NotTo(gomega.BeEmpty())
+
+		framework.Logf("Creating Deployment")
+		labelsMap := make(map[string]string)
+		labelsMap["app"] = "test"
+		deployment, err := createDeployment(
+			ctx, client, 1, labelsMap, nil, namespace, pvclaims, "", false, busyBoxImageOnGcr)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		deployment, err = client.AppsV1().Deployments(namespace).Get(ctx, deployment.Name, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		pods, err := fdep.GetPodsForDeployment(client, deployment)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		pod := pods.Items[0]
+		err = fpod.WaitForPodNameRunningInNamespace(client, pod.Name, namespace)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		return statefulset, deployment, volumesBeforeScaleDown
+	}
+
+	return statefulset, nil, volumesBeforeScaleDown
+}
+
+// volumeLifecycleActions creates pvc and pod and waits for them to be in healthy state and then deletes them
+func volumeLifecycleActions(ctx context.Context, client clientset.Interface, namespace string,
+	sc *storagev1.StorageClass) {
+	pvc1, err := createPVC(client, namespace, nil, diskSize, sc, "")
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	pvs, err := fpv.WaitForPVClaimBoundPhase(
+		client, []*v1.PersistentVolumeClaim{pvc1}, framework.ClaimProvisionTimeout)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	volHandle := pvs[0].Spec.CSI.VolumeHandle
+
+	pod1, err := createPod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvc1}, false, execCommand)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	vmUUID := getNodeUUID(ctx, client, pod1.Spec.NodeName)
+	framework.Logf("VMUUID : %s", vmUUID)
+	isDiskAttached, err := e2eVSphere.isVolumeAttachedToVM(client, volHandle, vmUUID)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	gomega.Expect(isDiskAttached).To(gomega.BeTrue(),
+		"Volume is not attached to the node volHandle: %s, vmUUID: %s", volHandle, vmUUID)
+
+	framework.Logf("Verify the volume is accessible")
+	_, err = framework.LookForStringInPodExec(namespace, pod1.Name,
+		[]string{"/bin/cat", "/mnt/volume1/fstype"}, "", time.Minute)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	err = fpod.DeletePodWithWait(client, pod1)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	deletePodAndWaitForVolsToDetach(ctx, client, pod1)
```

Comment:
this routing actually calls `DeletePodWithWait()` internally, so no need to call `DeletePodWithWait()` in line 660